proposals/0xxx-enum-cases-as-protocol-witnesses.md

```diff
@@ -0,0 +1,211 @@
+
+You will still be able to implement the requirement manually if you want and code that currently compiles today (with the manual implementation) will continue to compile. However, you will now have the option to let the case satisfy the requirement directly.
+
+Here are a few more examples that demonstrate how cases will be matched with the requirements:
+
+```swift
+protocol Foo {
+  static var zero: FooEnum { get }
+  static var one: Self { get }
+  static func two(arg: Int) -> FooEnum
+  static func three(_ arg: Int) -> Self
+  static func four(_ arg: String) -> Self
+  static var five: Self { get }
+  static func six(_: Int) -> Self
+  static func seven(_ arg: Int) -> Self
+  static func eight() -> Self
+}
+
+enum FooEnum: Foo {
+  case zero // okay
+  case one // okay
+  case two(arg: Int) // okay
+  case three(_ arg: Int) // okay
+  case four(arg: String) // not a match
+  case five(arg: Int) // not a match
+  case six(Int) // okay
+  case seven(Int) // okay
+  case eight // not a match
+}
+```
+
+The last one is intentional - there is no way to declare a `case eight()` today (and even when you could in the past, it actually had a different type). In this case, the requirement `static func eight()` can infact be better expressed as a `static var eight`. In the future, this limitation may be lifted when other kinds of witness matching is considered.
+
+## Source compatibility
+
+This does not break source compatibility since it's a strictly additive change and allows code that previously did not compile to now compile and run successfully.
+
+## Effect on ABI stability
+
+This does not affect the ABI and does not require new runtime support.
+
+## Effect on API resilience
+
+Library authors can freely switch between having the enum case satisfy the requirement or implementing the requirement directly (and returning a different case). However, while the return _type_ will stay the same, the return _value_ will be different in both cases and so it can break code that is relying on the exact returned value.
+
+For example:
+
+```swift
+protocol Foo {
+  static var bar: Self { get }
+}
+
+enum FooEnum: Foo { // #1
+  case bar
+}
+
+enum FooEnum: Foo { // #2
+  case _bar
+  static var bar: Self { return ._bar }
+}
+
+func takesFoo<T: Foo>(value: T) { ... }
+let direct = FooEnum.bar // #1
+let asProperty = FooEnum.bar // #2
+
+takesFoo(value: direct) // This one recieves .bar
+takesFoo(value: asProperty) // This one recieves ._bar
+```
+
+## Alternatives considered
+
+- Allow protocol requirements to be declared as `case` instead of `static var` or `static func` - the obvious downside of doing this would be that only enums would be able to adopt such a protocol, which would be unreasonably restrictive beacuse other types like classes and structs having satisfying witnesses would no longer be able to adopt such a protocol.
```

Comment:
`beacuse` ➡️ `because`

```suggestion
- Allow protocol requirements to be declared as `case` instead of `static var` or `static func` - the obvious downside of doing this would be that only enums would be able to adopt such a protocol, which would be unreasonably restrictive because other types like classes and structs having satisfying witnesses would no longer be able to adopt such a protocol.
```